standalone-shopify-template/web/components/App.jsx

```diff
@@ -0,0 +1,148 @@
+import { api } from "../api";
+import Index from "../routes/index";
+import SignedInPage from "../routes/signed-in";
+import SignInPage from "../routes/sign-in";
+import SignUpPage from "../routes/sign-up";
+import ResetPasswordPage from "../routes/reset-password";
+import VerifyEmailPage from "../routes/verify-email";
+import ChangePassword from "../routes/change-password";
+import ForgotPassword from "../routes/forgot-password";
+import "./App.css";
+import { AuthProvider, ShopProvider } from "../providers";
+
+const App = () => {
+  useEffect(() => {
+    document.title = `${process.env.GADGET_APP}`;
+  }, []);
+
+  const router = createBrowserRouter(
+    createRoutesFromElements(
+      // Changed the index route to dashboard to avoid an infinite redirect loop
+      <Route path="/" element={<Layout />}>
+        <Route
+          index
+          path="dashboard"
+          element={
+            <SignedOutOrRedirect path="signed-in">
+              <Index />
+            </SignedOutOrRedirect>
+          }
+        />
+        <Route
+          path="signed-in"
+          element={
+            <SignedInOrRedirect>
+              <AuthProvider>
+                <ShopProvider>
+                  <SignedInPage />
+                </ShopProvider>
+              </AuthProvider>
+            </SignedInOrRedirect>
+          }
+        />
+        <Route
+          path="change-password"
+          element={
+            <SignedInOrRedirect>
+              <ChangePassword />
+            </SignedInOrRedirect>
+          }
+        />
+        <Route
+          path="forgot-password"
+          element={
+            <SignedOutOrRedirect>
+              <ForgotPassword />
+            </SignedOutOrRedirect>
+          }
+        />
+        <Route
+          path="sign-in"
+          element={
+            <SignedOutOrRedirect>
+              <SignInPage />
+            </SignedOutOrRedirect>
+          }
+        />
+        <Route
+          path="sign-up"
+          element={
+            <SignedOutOrRedirect>
+              <SignUpPage />
+            </SignedOutOrRedirect>
+          }
+        />
+        <Route path="reset-password" element={<ResetPasswordPage />} />
+        <Route path="verify-email" element={<VerifyEmailPage />} />
+      </Route>
+    )
+  );
+
+  return (
+    <Suspense fallback={<></>}>
+      <RouterProvider router={router} />
+    </Suspense>
+  );
+};
+
+const Layout = () => {
+  return (
+    // Provider set to `AppType.Standalone` so tht it doesn't try to embed the app in the Shopify admin
+    <Provider
+      type={AppType.Standalone}
+      shopifyApiKey={window.gadgetConfig.apiKeys.shopify}
+      api={api}
+    >
+      <Header />
+      <div className="app">
+        <div className="app-content">
+          <div className="main">
+            <Outlet />
+          </div>
+        </div>
+      </div>
+    </Provider>
+  );
+};
+
+const Header = () => {
+  return (
+    <div className="header">
```

Comment:
Same comment, should we be using Polaris?

Reply:
Do you think that I should use MUI instaed?

Reply:
The code there is basically the stuff we give in the general auth code + some additions/reworks